hack/olm/catalog-src.yaml

```diff
@@ -16,6 +16,12 @@ spec:
   sourceType: grpc
   grpcPodConfig:
     securityContextConfig: restricted
+    nodeSelector:
+      node-role.kubernetes.io: infra
```

Comment:
@anispate , the PR says `prefer` instead of enforce. Is that really the case? I haven't tested this but my hunch is that this enforces Catalog pods to be scheduled only on infra nodes, doesn't it?

Reply:
It was already tested before by us for other operators.

Reply:
If it enforces, like mentioned in the slack thread, we wouldn't want to enforce this on all users of the catalog. 
cc: @jan--f

Reply:
Agreed, this configuration enforces that it can only be scheduled onto infra nodes. We likely need a separate CatalogSource for OSD/ROSA compared to the generic one